src/ruleset.rs

```diff
@@ -491,14 +491,32 @@ impl RulesetCreated {
         }
     }
 
+    /// Creates a copy of this object and duplicates the file descriptor with
+    /// CLOEXEC set.
+    ///
+    /// On error, returns a wrapped [`RulesetError::Duplicate`].
+    pub fn duplicate(&self) -> Result<RulesetCreated, RulesetError> {
```

Comment:
`std::io::File` calls this `try_clone`.

Reply:
The issue with the `clone` name is that developers would think it is a standalone object, which is not the case because the kernel object, referenced by the ruleset FD, would be the same. This means that the one of these object can affect the other.

This `duplicate` method should probably be renamed to something like `new_ref()`, and be part of the `RulesetCreatedAttr` trait. I'm wondering if it wouldn't be safer to keep a reference to the original object to avoid unexpected changes. :thinking:  Would that be an issue for your use case?

Reply:
No it wouldn't, I'll work on those changes.